packages/cli/src/lib/plugin.js

```diff
@@ -0,0 +1,250 @@
+import fs from 'fs'
+import path from 'path'
+
+import chalk from 'chalk'
+
+import { getCompatibilityData } from '@redwoodjs/cli-helpers'
+
+import { installModule, isModuleInstalled } from './packages'
+
+import { getPaths } from './index'
+
+/**
+ * The file inside .redwood which will contain cached plugin command mappings
+ */
+const PLUGIN_CACHE_FILENAME = 'commandCache.json'
+
+/**
+ * A cache of yargs information for redwood commands that are available from plugins.
+ *
+ * This is intended to be used for commands which lazy install their dependencies so that
+ * this information otherwise would not be available and help output would be unavailable/
+ * incorrect.
+ */
+export const PLUGIN_CACHE_DEFAULT = {
+  '@redwoodjs/cli-storybook': {
+    storybook: {
+      aliases: ['sb'],
+      description:
+        'Launch Storybook: a tool for building UI components and pages in isolation',
+    },
+  },
+  '@redwoodjs/cli-data-migrate': {
+    'data-migrate <command>': {
+      aliases: ['dataMigrate', 'dm'],
+      description: 'Migrate the data in your database',
+    },
+  },
+}
+
+/**
+ * A list of commands that are built into the CLI and require no plugin to be loaded.
+ */
+export const PLUGIN_CACHE_BUILTIN = [
+  'build',
+  'check',
+  'diagnostics',
+  'console',
+  'c',
+  'deploy',
+  'destroy',
+  'd',
+  'dev',
+  'exec',
+  'experimental',
+  'exp',
+  'generate',
+  'g',
+  'info',
+  'lint',
+  'prerender',
+  'render',
+  'prisma',
+  'record',
+  'serve',
+  'setup',
+  'test',
+  'ts-to-js',
+  'type-check',
+  'tsc',
+  'tc',
+  'upgrade',
+]
+
+export function loadCommadCache() {
+  // Always default to the default cache
+  let pluginCommandCache = PLUGIN_CACHE_DEFAULT
+  try {
+    const localCommandCache = JSON.parse(
+      fs.readFileSync(
+        path.join(getPaths().generated.base, PLUGIN_CACHE_FILENAME)
+      )
+    )
+    // This validity check is rather naive but it exists to invalidate a
+    // previous format of the cache file
+    let valid = true
+    for (const [key, value] of Object.entries(localCommandCache)) {
+      if (key === '_builtin') {
+        continue
+      }
+      valid &&= !Array.isArray(value)
+    }
+    if (valid) {
+      // Merge the default cache with the local cache but ensure the default
+      // cache takes precedence - this ensure the cache is consistent with the
+      // current version of the framework
+      pluginCommandCache = {
+        ...localCommandCache,
+        ...PLUGIN_CACHE_DEFAULT,
+      }
+    }
+  } catch (error) {
+    // If the cache file doesn't exist we can just ignore it and continue
+    if (error.code !== 'ENOENT') {
+      console.error(error)
```

Comment:
i know we don't have a convention here, but this error could be better; we should at least say something like "error loading plugin command cache", and the filename (you'd have to make the `localCommandCacheFile` var earlier up in the file):

```suggestion
      console.error(
        `Error loading plugin command cache at ${localCommandCacheFile}`
      )
      console.error(error)
```

e.g., it's hard to see where the error's coming from in this stack trace:

```
redwood-app % yarn rw info
SyntaxError: Unexpected token } in JSON at position 795
    at parse (<anonymous>)
    at JSON.parse (/Users/dom/projects/redwood/redwood-app/node_modules/core-js/modules/esnext.json.parse.js:249:56)
    at loadCommadCache (/Users/dom/projects/redwood/redwood-app/node_modules/@redwoodjs/cli/dist/lib/plugin.js:65:36)
    at loadPlugins (/Users/dom/projects/redwood/redwood-app/node_modules/@redwoodjs/cli/dist/plugin.js:44:58)
    at runYargs (/Users/dom/projects/redwood/redwood-app/node_modules/@redwoodjs/cli/dist/index.js:183:33)
    at /Users/dom/projects/redwood/redwood-app/node_modules/@redwoodjs/cli/dist/index.js:137:13
    at AsyncLocalStorage.run (node:async_hooks:338:14)
    at AsyncLocalStorageContextManager.with (/Users/dom/projects/redwood/redwood-app/node_modules/@opentelemetry/sdk-trace-node/node_modules/@opentelemetry/context-async-hooks/build/src/AsyncLocalStorageContextManager.js:33:40)
    at ContextAPI.with (/Users/dom/projects/redwood/redwood-app/node_modules/@opentelemetry/api/build/src/api/context.js:60:46)
    at Tracer.startActiveSpan (/Users/dom/projects/redwood/redwood-app/node_modules/@opentelemetry/sdk-trace-node/node_modules/@opentelemetry/sdk-trace-base/build/src/Tracer.js:122:32)

  System:
    OS: macOS 13.4.1
    Shell: 5.9 - /bin/zsh
  Databases:
    SQLite: 3.39.5 - /usr/bin/sqlite3
  Browsers:
    Chrome: 117.0.5938.62
    Safari: 16.5.2
  npmPackages:
    @redwoodjs/cli-storybook: 6.2.2 => 6.0.7 
    @redwoodjs/core: 6.2.2 => 6.0.7 
```

Reply:
Yeah I should have done this from the start - agreed. I'd like to sit down at some point and document some sort of CLI style guide and then implement the appropriate helpers it. I tend to just not want to get into the weeds with exactly what to communicate and how. 

I'll update to output exactly what you suggest.